pkg/db/db_client/db_client.go

```diff
@@ -80,9 +84,15 @@ func NewDbClient(ctx context.Context, connectionString string, onConnectionCallb
 		return nil, err
 	}
 
+	// load up the server settings
+	if err := client.loadServerSettings(ctx); err != nil {
+		client.Close(ctx)
```

Comment:
maybe we should have a defer which checks the error and closes the client if the client is non nil and the error is not nil